components/brave_shields/content/browser/domain_block_navigation_throttle.cc

```diff
@@ -48,13 +65,32 @@ std::pair<bool, std::string> ShouldBlockDomainOnTaskRunner(
       url, blink::mojom::ResourceType::kMainFrame, url.host(),
       aggressive_for_engine, false, false, false);
 
-  const bool should_block =
+  block_result.should_block =
       result.important || (result.matched && !result.has_exception);
 
-  std::string new_url = aggressive_setting && result.rewritten_url.has_value
-                            ? std::string(result.rewritten_url.value)
-                            : std::string();
-  return std::pair(should_block, new_url);
+  block_result.new_url = aggressive_setting && result.rewritten_url.has_value
+                             ? std::string(result.rewritten_url.value)
+                             : std::string();
+
+  if (block_result.should_block || result.has_exception) {
+    content::devtools_instrumentation::AdblockInfo info;
+    info.request_url = url;
+    info.checked_url = url;
+    info.source_host = url.host();
+    info.resource_type = blink::mojom::ResourceType::kMainFrame;
+    info.aggressive = aggressive_for_engine;
+    info.blocked = true;
```

Comment:
nit: one of the conditions for this `if` is `has_exception`, so `info.blocked` should not necessarily always be true. We do have `info.did_match_exception` so what happened could be inferred, but it might be confusing.